lib/dor/models/identifiable.rb

```diff
@@ -233,7 +233,20 @@ def adapt_to_cmodel
       if object_class
         self.instance_of?(object_class) ? self : self.adapt_to(object_class)
       else
```

Comment:
what if `identityMetadata` or `identityMetadata.objectType` are `nil`? (i.e., identityMetadata datastream is missing or lacking an objectType element. won't that throw a `NoMethodError` and cause `adapt_to_cmodel` to raise out rather than executing this block here.